docs/developer/designs/min-runtime/README.md

```diff
@@ -0,0 +1,118 @@
+
+## Motivation
+
+Unrestrained preemption can lead to resource thrashing, where jobs are repeatedly preempted before making meaningful progress. The min-runtime feature addresses this issue by providing configurable minimum runtime guarantees that can be set by either cluster operators or sometimes within parts of the queue to provide guarantees about minimal useful work.
+
+## Detailed Design
+
+### Reclaim Min Runtime Configuration
+
+The reclaim-min-runtime parameter can be configured with the following values:
+
+- **0 (default)**: Jobs are always preemptible via reclaims
+- **-1**: Jobs are never preemptible via reclaims
+- **Positive value (in seconds)**: Minimum guaranteed runtime before preemption via reclaims
+
+### Preempt Min Runtime Configuration
+
+In addition to protecting jobs from being preempted too early with reclaim-min-runtime, we also introduce preempt-min-runtime to ensure that in-queue preemptions are also protected with min-runtime.
+
+The preempt-min-runtime parameter can be configured with the following values:
+
+- **0 (default)**: Jobs can always be preempted by others (subject to reclaim-min-runtime constraints) in the same queue
+- **-1**: Jobs can never be preempted by others in the same queue
+- **Positive value (in seconds)**: Minimum guaranteed runtime before in-queue preemption
+
+### Configuration Hierarchy
+
+The configuration follows a hierarchical override structure:
+
+1. **Node Pool Level**: Base configuration that applies to all reclaims/preemptions if not overridden by a queue. Default will be set to 0 which preserves existing behaviors to always reclaim/preempt.
+2. **Queue Level**: Overrides node pool configuration for reclaims/preemptions, can be further overridden by a child queue. Default will be set to unassigned, causing the node pool level value to be used.
+
+### Resolving the applicable min-runtime for reclaims and preemptions
+
+#### Reclaims (preemptor and preemptee are in different queues)
+1. Resolve the lowest common ancestor between the leaf-queues of preemptor and preemptee.
+2. From the LCA, take 1 step in the hierarchy towards the preemptee's leaf queue
+3. Use the reclaim-min-runtime from this queue, if it is set, otherwise move back up towards root of tree and default to node pool-level configuration value.
+
+The idea around the algorithm here is to isolate settings of min-runtime in the queue tree to only affect siblings in reclaim scenarios, and for the potential to distribute the administration of these values in the queue tree (such as giving a user access to change parts of the tree). 
+As a follow-up, we could also provide a setting to disable this and always use the leaf-tree resolved value in all cases. This could be favorable in a scenario where all min-runtimes in the queue tree are managed by one entity.
+
+##### Example
+1. Leaf queues `root.A.B.C.leaf1` (preemptor) and `root.A.B.D.leaf2` (preemptee) will use the min-runtime resolved for `root.A.B.D`.
```

Comment:
I think we need to elaborate more here. Specify that the "leafs" are jobs (if I understood correctly) and that Leaf1 is attempting to reclaim from Leaf2. Also, let's add one example where the sibling queues are higher level, say: root.A.B.C is trying to reclaim from root.A.D.E, and explain that the value will be the min reclaim of D (or whatever default value D inherited from it's ancestors)

Reply:
will do, the leaf1 and leaf2 were actually queues here and it is loosely defined as jobs scheduled under these queues. Will clarify.